spec/requests/session_controller_spec.rb

```diff
@@ -808,18 +808,113 @@ def sso_for_ip_specs
       logged_on_user = User.find(logged_on_user.id)
 
       expect(logged_on_user.admin).to eq(false)
-      expect(logged_on_user.email).to eq('bob@bob.com')
-      expect(logged_on_user.name).to eq('Sam Saffron')
-      expect(logged_on_user.username).to eq('sam')
+      expect(logged_on_user.email).to eq("bob@bob.com")
+      expect(logged_on_user.name).to eq("Sam Saffron")
+      expect(logged_on_user.username).to eq("sam")
 
       expect(logged_on_user.single_sign_on_record.external_id).to eq("666")
-      expect(logged_on_user.single_sign_on_record.external_username).to eq('sam')
+      expect(logged_on_user.single_sign_on_record.external_username).to eq("sam")
       expect(logged_on_user.active).to eq(true)
       expect(logged_on_user.custom_fields["shop_url"]).to eq("http://my_shop.com")
       expect(logged_on_user.custom_fields["shop_name"]).to eq("Sam")
       expect(logged_on_user.custom_fields["bla"]).to eq(nil)
     end
 
+    context "when an invitation is used" do
+      let(:invite) { Fabricate(:invite, email: invite_email, invited_by: Fabricate(:admin)) }
+      let(:invite_email) { nil }
+
+      def login_with_sso_and_invite(invite_key = invite.invite_key)
+        write_secure_session("invite-key", invite_key)
+        sso = get_sso("/")
+        sso.external_id = "666" # the number of the beast
```

Comment:
Is the comment required here?

Reply:
It is just copied and pasted from another spec here, see all the others in the file. Will remove them

Reply:
Lol now I'm curious who wrote them